`undefined reference to` errors

I do have glm, libxext, etc. installed. also I'm using a source distro so the header files are there...
This is with the newest version btw.

```
libslopy.so: undefined reference to `XShapeCombineRectangles'
libslopy.so: undefined reference to `XScreenNumberOfScreen'
libslopy.so: undefined reference to `XGetWindowProperty'
libslopy.so: undefined reference to `XGrabPointer'
libslopy.so: undefined reference to `XQueryTree'
libslopy.so: undefined reference to `XCreateFontCursor'
libslopy.so: undefined reference to `XSetClassHint'
libslopy.so: undefined reference to `XKeysymToKeycode'
libslopy.so: undefined reference to `XCreateColormap'
libslopy.so: undefined reference to `glXDestroyContext'
libslopy.so: undefined reference to `XInternAtom'
libslopy.so: undefined reference to `XAllocWMHints'
libslopy.so: undefined reference to `glXMakeCurrent'
libslopy.so: undefined reference to `XAllocColor'
libslopy.so: undefined reference to `XChangeProperty'
libslopy.so: undefined reference to `XUngrabPointer'
libslopy.so: undefined reference to `glXGetProcAddressARB'
libslopy.so: undefined reference to `XFreeCursor'
libslopy.so: undefined reference to `XMapWindow'
libslopy.so: undefined reference to `XQueryPointer'
libslopy.so: undefined reference to `XQueryKeymap'
libslopy.so: undefined reference to `glXCreateContext'
libslopy.so: undefined reference to `XGrabKeyboard'
libslopy.so: undefined reference to `XFree'
libslopy.so: undefined reference to `XUngrabKeyboard'
libslopy.so: undefined reference to `XMatchVisualInfo'
libslopy.so: undefined reference to `glXWaitGL'
libslopy.so: undefined reference to `XOpenDisplay'
libslopy.so: undefined reference to `XSetWMProperties'
libslopy.so: undefined reference to `XFlush'
libslopy.so: undefined reference to `XCreateWindow'
libslopy.so: undefined reference to `XDestroyWindow'
libslopy.so: undefined reference to `XTranslateCoordinates'
libslopy.so: undefined reference to `XChangeActivePointerGrab'
libslopy.so: undefined reference to `XCheckTypedEvent'
libslopy.so: undefined reference to `XGetWindowAttributes'
libslopy.so: undefined reference to `glXSwapBuffers'
libslopy.so: undefined reference to `XGetSelectionOwner'
libslopy.so: undefined reference to `XCloseDisplay'
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,6 +47,8 @@ set_property(TARGET ${LIBRARY_NAME} PROPERTY CXX_STANDARD_REQUIRED ON)
 add_executable(${EXECUTABLE_NAME} src/options.cpp
                                   src/main.cpp)
 
+set_property(TARGET ${EXECUTABLE_NAME} PROPERTY CXX_STANDARD 11)
+
 set( CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/modules" )
 
 find_package(GLM REQUIRED)
@@ -61,12 +63,13 @@ include_directories(${X11_INCLUDE_DIR}
                     ${GLX_INCLUDE_DIR}
                     ${OPENGL_INCLUDE_DIR})
 
-target_link_libraries(${EXECUTABLE_NAME} ${X11_LIBRARIES}
-                                         ${GLM_LIBRARIES}
-                                         ${OPENGL_LIBRARIES}
-                                         ${GLX_LIBRARY}
-                                         ${XEXT_LIBRARIES}
-                                         ${LIBRARY_NAME})
+target_link_libraries(${LIBRARY_NAME} ${X11_LIBRARIES}
+                                      ${GLM_LIBRARIES}
+                                      ${OPENGL_LIBRARIES}
+                                      ${GLX_LIBRARY}
+                                      ${XEXT_LIBRARIES})
+
+target_link_libraries(${EXECUTABLE_NAME} ${LIBRARY_NAME})
 
 install( TARGETS ${EXECUTABLE_NAME} DESTINATION "${CMAKE_INSTALL_PREFIX}/bin" )
 install( TARGETS ${LIBRARY_NAME} DESTINATION "${CMAKE_INSTALL_PREFIX}/lib" )

